arkouda/numeric.py

```diff
@@ -474,21 +478,52 @@ def hash(
     cancel each other out, hence we do a rotation by the ordinal of
     the array.
     """
-    if isinstance(pda, pdarray):
-        return _hash_single(pda, full)
-
-    repMsg = type_cast(
-        str,
-        generic_msg(
-            cmd="efuncArr",
-            args={
-                "nameslist": [n.name for n in pda],
-                "typeslist": [n.objType for n in pda],
-                "length": len(pda),
-                "size": len(pda[0]),
-            },
-        ),
-    )
+    from arkouda import Categorical as _Categorical
+    from arkouda import SegArray as _SegArray
+
+    if isinstance(pda, pdarray) or isinstance(pda, _Categorical) or isinstance(pda, _SegArray):
+        if isinstance(pda, _Categorical):
+            a = pda.codes
+        elif isinstance(pda, _SegArray):
+            a = pda.values
```

Comment:
also i know i signed off hashing the values for segarray during initial design decisions, but after playing around a bit more, I don't think this is what we want to do. I think ultimately segarrays are gonna be difficult to handle (for reasons im about to get into). I think we're gonna wanna do some brainstorming for them, but they are def outside the scope of this bug fix and we should put in a separate issue to handle them

```python
# consider this segarray
>>> sa = ak.segarray(ak.array([0,3,6]), ak.array([0,1,2,3,4,5,0,1,2]))
>>> sa
SegArray([
[0 1 2]
[3 4 5]
[0 1 2]
])

# intuitively the first and third elem should has to the same thing
# since they are the same array and the second should be it's own thing

# we can't get this behavior by hashing the segments since 
# they don't capture the fact that sa[0] and sa[2] are identical
>>> sa.segments
array([0 3 6])

>>> ak.hash(sa.segments)
(array([14020738532124324502 10334800848021183012 12294710806803928155]),
 array([13660161615634858013 14420950124940837019 318151776295263894]))

# we can't hash the values cause they don't really match back up to the original segarray
>>> ak.hash(sa.values)
(array([14020738532124324502 17469661437632883388 2223077377302025035 10334800848021183012 12899154897215201871 1827545583036225533 14020738532124324502 17469661437632883388 2223077377302025035]),
 array([13660161615634858013 100355499037336788 7178333632624491744 14420950124940837019 4776206602043303594 14424044815130752497 13660161615634858013 100355499037336788 7178333632624491744]))

# we thought about hashing each sa[i] and then hashing the resulting pdarray
# but ak.hash(pdarray) returns a pdarray not a single value

# we can't just concatenate the values within a segment because how do u differentiate between 
# [1, 11, 2] and [11, 1, 2] for instance

# one absolutely awful idea that technically works... is to interpret each inner array as a string and then hash that
# (shown here using __repr__ but we would have to do cast server side because repr truncates)
>>> ak.hash(ak.array([ak.array(sa[i]).__repr__() for i in range(3)]))
(array([16660294113708371871 6932019251617944032 16660294113708371871]),
 array([10308859934083080982 18056049045803087039 10308859934083080982]))
# we see that the first and last do match, but it's such an ugly solution
# If we go this route maybe we do hex representation to shorten string lengths
```
These are just some quick ideas, but doing this well is gonna require some thought. I vote we hold off on segarray hashing for now

Reply:
This has been captured in #2459

Reply:
Removed SegArray handling from this PR